hudi-flink-datasource/hudi-flink1.14.x/src/main/java/org/apache/hudi/table/format/cow/ParquetSplitReaderUtil.java

```diff
@@ -460,59 +460,59 @@ private static WritableColumnVector createWritableColumnVector(
       case BOOLEAN:
         checkArgument(
             typeName == PrimitiveType.PrimitiveTypeName.BOOLEAN,
-            "Unexpected type: %s", typeName);
+            "Unexpected type exception. Primitive type: %s. Field type: %s.", typeName, fieldType.getTypeRoot().name());
```

Comment:
Can we abstract it as an utility?

Reply:
I extracted it as a static method. The code for error message construction won't be duplicated too many times.